### ISSUE TITLE
Translate 2019 content into Korean

### ISSUE THREAD

**Issue by bazzadp** [HTTPArchive/almanac.httparchive.org#926]
These are the chapters to be translated, and they exist in the [content directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content):
- [ ] accessibility - 
- [ ] caching - 
- [ ] cdn - 
- [ ] cms - 
- [ ] compression - 
- [ ] css - 
- [ ] ecommerce - 
- [ ] fonts - 
- [ ] http2 - 
- [ ] javascript - 
- [ ] markup - 
- [ ] media - 
- [ ] mobile-web - 
- [ ] page-weight - 
- [ ] performance - 
- [ ] pwa - 
- [ ] resource-hints - 
- [ ] security - 
- [ ] seo - 
- [ ] third-parties - 

Please include "Makes progress on #539" in all pull requests so a link is created from the PR to this issue.

Additionally the following pages need translated too in the [language specific templates directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/templates):
- [ ] base.html
- [ ] base_chapter.html
- [ ] contributors.html
- [ ] error.html
- [ ] index.html
- [ ] featured_chapters.html
- [ ] methodology.html 
- [ ] table_of_contents.html 
- [ ] accessibility_statement.html 
- [ ] base_ebook.html

There is no need to translate the chapters HTML pages are they are generated off the markdown combined with the above templates.

---

Common notes for writing consistency are here: https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide. Feel free to edit that and/or add Korean-specific extras by editing this comment.

Korean specific extra advice:

*

**Comment by bazzadp** [comment]
You really need these to be able to run the site in a new language so suggest you start with them first:

- base.html
- base_chapter.html
- index.html
- table_of_contents.html

They are also a lot easier to translate than the chapters themselves as a lot shorter!

then up to you whether you finish out all the templates or tackle a chapter.

**Comment by bazzadp** [comment]
You'll also need to add Korean to here:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/bc835a83ec00489f36253d5751dea4a7267ca703/src/language.py#L28-L33

and the language code (ko?) to here (but let's not commit this second one until we have at least the main templates and a chapter or two done because committing it here means it would be put live on next release):

https://github.com/HTTPArchive/almanac.httparchive.org/blob/bc835a83ec00489f36253d5751dea4a7267ca703/src/config/2019.json#L5

And while when you submit your first PR, add yourself to the contributors section of the config under your GitHub id, which means you'll show up in the [2019 Contributors page](https://almanac.httparchive.org/en/2019/contributors?teams=translators):

https://github.com/HTTPArchive/almanac.httparchive.org/blob/bc835a83ec00489f36253d5751dea4a7267ca703/src/config/2019.json#L181-L189

**Comment by rviscomi** [comment]
@techhtml I'd like to post something from the HTTP Archive Twitter account to promote this translation effort and hopefully find more translators to help with it. Could you translate this text for the tweet?

> Do you speak Korean? Help us translate the Web Almanac as part of our effort to make it globally accessible to the web community!

Thanks for your help!

**Comment by goleedev** [comment]
@rviscomi If you still need some help, here is the translation! 🙂

EN: Do you speak Korean? Help us translate the Web Almanac as part of our effort to make it globally accessible to the web community!
KO: 한국어를 할 줄 아시나요? 그렇다면 웹 커뮤니티를 전 세계에서 액세스할 수 있도록 Web Almanac의 번역을 도와 주세요!

**Comment by rviscomi** [comment]
Thanks @goleedev! I've just tweeted it here: https://twitter.com/HTTPArchive/status/1293965549588877314

**Comment by taechonkim** [comment]
Hi, is this issue valid now? if so, I want to contribute it! how can I contribute?

**Comment by bazzadp** [comment]
Hey @taechonkim would love to have your help. Unfortunately we've not made much progress on this so far.

I've just sent you an invite to the team. After you accept that, you should be able to edit the [first comment](https://github.com/HTTPArchive/almanac.httparchive.org/issues/926#issue-646692152) and put your name besides some of the files to translate to let others know you're working on that one.

Please do them in the order listed (so start base.html and finish with methodology.html). Then just fork the repo, copy the English file to a `ko` folder and translate it! Then submit a PR with that.

We've a [Translator's Guide](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide) and also a [Git Guide](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Git-Guide) that are worth a read before you start!

Let me know if you have any questions.

**Comment by taechonkim** [comment]
Hey @bazzadp I have a question.
I have forked the original repository, cloned forked repository to local, and made branch for my translation job. 
I think my next job is copying src/templates/en to src/templates/ko, starting translation with base.html. is it right?

**Comment by bazzadp** [comment]
Correct. Only copy across the templates as you translate them so you don't commit any English version ones into src/templates/ko. You can decide how many translated files to include in each pull request. base.html is quite big so maybe do one PR for that, for example. Another example the 2019 and 2020 contributor pages are obviously very similar so can submit both in the same PR. Can also do all the templates at once if you want, but smaller PRs are easier to review and merge so recommend that!

**Comment by taechonkim** [comment]
Hey @bazzadp Thank you for quick reply. 
I translating base.html now, there are details about contributors bottom part of base.html (about 114 line, starting with details of kleinnab...)
I wonder where these details comes with in Web Almanac. 
And is there need to translate them?

**Comment by bazzadp** [comment]
I think you mean line 179 not 114? Try to keep the line numbering consistent with English as makes it easier to manage.

These are author bios and they are given at the bottom of the chapters they wrote ([here's mine for example as I wrote the HTTP/2 chapter!](https://almanac.httparchive.org/en/2019/http2#authors)) so yes they do need translated too.

**Comment by hanj0415** [comment]
I've created a branch from the forked repository. I'm trying to start to translate base.html, but is it marked as completed from the job lists? If once the translation job is completed, what should I do next?

**Comment by tunetheweb** [comment]
No they were not completed so shouldn't be ticked. They were started upon but don't think we made much progress on them and didn't see a PR or branch with any of them done. Have unticked them now so work away!
